androidenhancedvideoplayer/src/main/java/com/profusion/androidenhancedvideoplayer/utils/TrackQualityItem.kt

```diff
@@ -0,0 +1,49 @@
+package com.profusion.androidenhancedvideoplayer.utils
+
+import androidx.compose.runtime.Stable
+import androidx.media3.common.TrackGroup
+import androidx.media3.common.util.UnstableApi
+
+sealed interface TrackQualityItem
+
+@UnstableApi
+@Stable
+class TrackQualityAuto(
+    private val label: String
+) : TrackQualityItem {
+    override fun toString(): String {
+        return label
+    }
+}
+
+@UnstableApi
+@Stable
+class TrackQualityItemValue(
+    val trackIndex: Int,
+    val group: TrackGroup
+) : Comparable<TrackQualityItemValue>, TrackQualityItem {
+
+    private val format get() = group.getFormat(trackIndex)
+
+    override fun compareTo(other: TrackQualityItemValue) = compareValuesBy(
+        this,
+        other,
+        { it.format.height },
+        { it.format.bitrate }
+    )
+
+    override fun equals(other: Any?): Boolean =
```

Comment:
Why don't we use `other == this`? If there are two `720p` quality items, they already come from different track indexes or come from different track groups so they must be treated differently

Reply:
I think that `==` is the same as `equals` in Kotlin, so won't that make this an infinity recursion?

Reply:
Got it. We can maintain as it is, then